### PR TITLE
fix: better examples on home

### PR DESCRIPTION
**Pull request by ovflowd**
This PR simply updates some examples to keep them easier to understand and easier to run/reproduce

### CODE REVIEW

**Review by canerakdas — APPROVED** [review]

**Review by MattIPv4 — COMMENTED** [review]
For the file names, thoughts on just have `// <name>.mjs` at the top of the file, and then `// Run with 'node <name>.mjs'` at the end of the file?

I think that feels a bit cleaner to me, and flows with the user journey of creating/reading the file, and then running it?

**Review by AugustinMauroy — APPROVED** [review]

**Review by shanpriyan — APPROVED** [review]

**Review by aymen94 — APPROVED** [review]

File: pages/en/index.mdx [hunk 1]
```diff
@@ -71,34 +73,37 @@ layout: home
...
     import { createHash } from 'node:crypto';
     import { readFile } from 'node:fs/promises';
 
     const hasher = createHash('sha1');
-    const fileContent = await readFile('./package.json');
 
     hasher.setEncoding('hex');
-    hasher.write(fileContent);
+    // ensure you have a `package.json` file for this test!
+    hasher.write(await readFile('./package.json'));
```
Comment (bmuenzenmeyer) [inline pages/en/index.mdx]: i tested this to make sure it works on windows pathing too

Comment (bmuenzenmeyer) [reply pages/en/index.mdx]: nit: the `./` is not necessary and is omitted in the Read Steams examples. In fact, it is what invited my concern about `path.sep` being needed or not

```suggestion
    hasher.write(await readFile('package.json'));
```

File: pages/en/index.mdx [hunk 2]
```diff
@@ -71,34 +73,37 @@ layout: home
...
     hasher.end();
 
     const fileHash = hasher.read();
```
Comment (bmuenzenmeyer) [inline pages/en/index.mdx:87]: nit: could it be more illustrative to have this console.log(hasher.read()) ???

**Review by bmuenzenmeyer — APPROVED** [review]
awesome to see you acheived better content while balancing the heights of all of them

small suggestion within, not blocking

File: pages/en/index.mdx [hunk 3]
```diff
@@ -71,34 +73,37 @@ layout: home
...
-
-    readableStream.on('data', chunk => writableStream.write(chunk));
+    import { createGzip } from 'node:zlib';
+    import { promisify } from 'node:util';
+
+    await promisify(pipeline)
+    ( // ensure you have a `package.json` file for this test!
+      createReadStream('package.json'),
+      createGzip(),
+      createWriteStream('package.json')
```
Comment (yoavain) [inline pages/en/index.mdx:101]: Not sure this makes sense. Shouldn't it be `package.json.gz`?

Comment (ovflowd) [reply pages/en/index.mdx:101]: That's correct.

**Comment by ovflowd** [comment]
> For the file names, thoughts on just have `// <name>.mjs` at the top of the file, and then `// Run with 'node <name>.mjs'` at the end of the file?
> 
> 
> 
> I think that feels a bit cleaner to me, and flows with the user journey of creating/reading the file, and then running it?

I saw this got ignored? @MattIPv4 feel free to make a PR! I was busy these days 🥲

File: pages/en/index.mdx (hunk 3 above)
Comment (ovflowd) [reply pages/en/index.mdx:101]: Fixed on main branch with a hotfix

**Pull request merged**